cli/azd/cmd/auth_token.go

```diff
@@ -81,7 +81,7 @@ func getTenantIdFromAzdEnv(
 	ctx context.Context,
 	envResolver environment.EnvironmentResolver,
 	subResolver account.SubscriptionTenantResolver) (tenantId string, err error) {
-	azdEnv, err := envResolver()
+	azdEnv, err := envResolver(ctx)
```

Comment:
Trying to test the feature and got this next error:

![image](https://github.com/Azure/azure-dev/assets/24213737/ad706a1e-bbe0-43c4-a236-41ca158388c4)


I created a new storage account and then set the ~/.azd/confing state like

```json
  "state": {
    "remote": {
      "backend": "AzureStorage",
      "config": {
        "accountName": "azdremotestatetest"
      }
    }
  },
```

Reply:
I ran `azd auth logout` and `azd auth login` and still getting same error

Reply:
User are required to have `Storage Blob Contributor` assigned to the storage account or container that they have configured.  @savannahostrowski @gkulin - we should make sure this point makes it into the docs/blog.